Src/FluentAssertions/Specialized/AssemblyAssertions.cs

```diff
@@ -139,6 +141,59 @@ public AndConstraint<AssemblyAssertions> Reference(Assembly assembly, string bec
         return new AndWhichConstraint<AssemblyAssertions, Type>(this, foundType);
     }
 
+    /// <summary>Asserts that the assembly is unsigned.</summary>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because"/>.
+    /// </param>
+    public AndConstraint<AssemblyAssertions> BeUnsigned(string because = "", params object[] becauseArgs)
+    {
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject.GetName().GetPublicKey() is not { Length: > 0 })
+            .FailWith(
+                "Expected assembly {0} to be unsigned{reason}, but it is.", Subject.FullName);
```

Comment:
🔧 We prefer to use the phrase "Did not expect the assembly {0} to be signed{reason}.". See also https://github.com/fluentassertions/fluentassertions/issues/1340